.github/workflows/dotnetcore.yml

```diff
@@ -0,0 +1,57 @@
+name: dotnetcore
+
+on:
+  push:
+    paths:
+      - "samples/dotnetcore/**"
+      - ".github/workflows/dotnetcore.yml"
+
+env:
+  DOCKER_REGISTRY: docker.pkg.github.com
+  USER: ${{ secrets.USER }}
+  AUTH_TOKEN: ${{ secrets.AUTH_TOKEN }}
+  OWNER: miracl
+
+jobs:
+  lint-dotnetcore:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+
+      - name: Set up DotNetCore 3.1
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: "3.1.100" # SDK Version to use.
+
+      - name: DotNetCore lint
+        run: dotnet build samples/dotnetcore/OidcSample.csproj
+
+  build-docker:
+    needs: lint-dotnetcore
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+
+      - name: Docker login, build & push
+        run: |
+          docker login -u $USER -p $AUTH_TOKEN $DOCKER_REGISTRY
+          docker build --tag $DOCKER_REGISTRY/$OWNER/oidc-samples/dotnetcore:latest samples/dotnetcore/
+          docker push $DOCKER_REGISTRY/$OWNER/oidc-samples/dotnetcore:latest
+          docker tag $DOCKER_REGISTRY/$OWNER/oidc-samples/dotnetcore:latest $DOCKER_REGISTRY/$OWNER/oidc-samples/dotnetcore:$GITHUB_ACTION
+          docker push $DOCKER_REGISTRY/$OWNER/oidc-samples/dotnetcore:$GITHUB_ACTION
+
+  run-integration-tests:
+    needs: build-docker
+    runs-on: ubuntu-latest
+    steps:
+      - name: Run integration tests
+        env:
+          CLIENT_ID: ${{ secrets.CLIENT_ID }}
+          CLIENT_SECRET: ${{ secrets.CLIENT_SECRET }}
+
+        run: |
+          docker login -u $USER -p $AUTH_TOKEN $DOCKER_REGISTRY
```

Comment:
I changed the way the file is formatted in my new PR, after a consultation with Vladi, so it's a bit more readable.
You can take a look, so you like and redo yours, because it's good that all files look the same.
https://github.com/miracl/oidc-samples/blob/29d5d52eeb12025f9640ab3b43de36c78b3a3975/.github/workflows/go.yml

Reply:
When I saw your PR, I saw I need to change that... Maybe it's better to do it before merging it to your PR.. so I will

Reply:
I think to update the dotnetcore workflow in another PR.. It is working now and would like to have some history